### PR TITLE
[backport v2.0] lib: updatehub: Minor bug fixes

### LINKED ISSUE #24211: [v2.2.x] lib: updatehub: Not working on Zephyr 2.x
**Describe the bug**
The FOTA UpdateHub stop to working on Zephyr 2.x

**To Reproduce**
Follow UpdateHub sample steps

**Expected behavior**
Be possible work with UpdateHub CE

**Impact**
Feature not work

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.11.2
 - v2.2.0

### PR DESCRIPTION
**Pull request by otavio**
This backport the fixes made in #23885.

fixes #24211

### CODE REVIEW

File: lib/updatehub/updatehub.c [hunk 1]
```diff
@@ -315,12 +322,40 @@ static int send_request(enum coap_msgtype msgtype, enum coap_method method,
 	return ret;
 }
 
+static bool install_update_cb_sha256(void)
+{
+	u8_t image_hash[TC_SHA256_DIGEST_SIZE];
+	char buffer[3], sha256_image_dowloaded[TC_SHA256_BLOCK_SIZE + 1];
```
Comment (d3zd3z) [inline lib/updatehub/updatehub.c:328]: We really need to fix this block size thing. This define has nothing to do with the length of the hex representation of the digest. This should be `TC_SHA256_DIGEST_SIZE*2 + 1`.

**Review by d3zd3z — CHANGES_REQUESTED** [review]

**Review by d3zd3z — CHANGES_REQUESTED** [review]
@otavio can you update this with the latest version that merged into master from #24154.

**Comment by otavio** [comment]
Sure. @nandojve can you handle it please?

**Comment by nandojve** [comment]
Added #24154

**Comment by carlescufi** [comment]
@ceolin @d3zd3z @jukkar this needs review in order to be transitioned to the ready column in the backports project

**Review by d3zd3z — APPROVED** [review]

**Review by ceolin — APPROVED** [review]
lgtm

**Comment by carlescufi** [comment]
@d3zd3z let's defer this one until we can identify the origin of all commits in this PR

**Comment by carlescufi** [comment]
@d3zd3z @ceolin please rebase to the latest 2.0-branch and build documentation locally to make sure it doesn't break it and report back please.

**Comment by d3zd3z** [comment]
One question is why there are 9 commits here, when it says it is a backport of another PR that only has 6 commits.

**Comment by nandojve** [comment]
>> One question is why there are 9 commits here, when it says it is a backport of another PR that only has 6 commits.

> d3zd3z requested changes on Apr 17
> d3zd3z left a comment
> @otavio can you update this with the latest version that merged into master from #24154.

The other 3 commits are from #24154.

**Comment by d3zd3z** [comment]
There are 9 commits. 6 are from #23885, and the other three are from #24154. Both have been merged.

**Comment by d3zd3z** [comment]
> @d3zd3z @ceolin please rebase to the latest 2.0-branch and build documentation locally to make sure it doesn't break it and report back please.

It is unclear to me what to do here @carlescufi. This is already based on the v2.0-branch, but I get documentation build failures when on v2.0-branch directly, even without these changes.

**Review by dleach02 — APPROVED** [review]

**Comment by nandojve** [comment]
We would like to have this changes on the v2.0 branch.
Could someone remove stale?

**Review by jukkar — APPROVED** [review]

**Comment by d3zd3z** [comment]
At this point 2.0 is beyond the support window, so this does not need to be merged, but can still be referenced externally.

**Pull request closed without merging**